src/util/mapbox.js

```diff
@@ -113,5 +129,13 @@ function parseUrl(url: string): UrlObject {
 
 function formatUrl(obj: UrlObject): string {
     const params = obj.params.length ? `?${obj.params.join('&')}` : '';
-    return `${obj.protocol}://${obj.authority}${obj.path}${params}`;
+    const url = `${obj.protocol}://${obj.authority}${obj.path}${params}`;
+    return transformURL(url);
 }
+
+exports.resolveURL = function(url: string): string {
+    url = transformURL(url);
+    const a = window.document.createElement('a');
+    a.href = url;
+    return transformURL(a.href);
```

Comment:
I think it might make sense to only do the URL transform *before* resolving it. This allows developers to use any URLs they want in the stylesheet, and requires them to be a resolvable URL only after they were transformed. E.g. on Native, we sometimes use custom URL *schemes* to indicate that this URL needs to be transformed. Going through the resolver before transformation means that they are going to be invalid, or at least in an unknown state.